### DISCUSSION TITLE
Kafka sink and dead letter queue with JSON schema/Avro

### DISCUSSION THREAD

**Discussion by chaporgin** [vectordotdev/vector#16925] [Q&A]
Hi! This question is a bit too broad and might not be related to Vector itself. I wanted to ask if having a JSON schema (or Avro) can I build dead letter queue for the pipeline
```
producer -> vector -> kafka
```
for the messages that do not comply to the schema.
I see that Vector supports [Avro schema](https://vector.dev/docs/reference/configuration/sinks/kafka/#encoding.avro.schema), but what happens if the messages cannot be serialized? There is no option to specify a specific secondary topic for such messages (IIUC). So what could be the options here? Thanks in advance!

**Comment by jszwedko** [comment]
Hey! Yeah, unfortunately events that don't match the schema will be dropped. We have an RFC for routing events that fail in sinks here: https://github.com/vectordotdev/vector/pull/14708. I think that would support your use-case to route those events that fail to be serialized to a secondary sink.
